x-pack/apm-server/sampling/processor_test.go

```diff
@@ -873,7 +873,10 @@ func TestDropLoop(t *testing.T) {
 
 				config.Elasticsearch = pubsubtest.Client(nil, nil) // disable pubsub
 
-				config.StorageLimit = 100 * 1024 // lower limit to trigger storage limit error
+				// lower limit to trigger storage limit error
+				// make it 5% higher than the existing size, and dropAndRecreate should kick in
+				// because there's a hardcoded 90% threshold for storage limit
+				config.StorageLimit = uint64(float64(lsm+vlog) * 1.05)
```

Comment:
[to reviewer] with this change in test, it fails before the fix and passes after.